operator/src/main/java/oracle/kubernetes/operator/http/HttpResponseStep.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.http;
+
+import java.net.HttpURLConnection;
+import java.net.http.HttpResponse;
+import java.util.Optional;
+
+import oracle.kubernetes.operator.work.Component;
+import oracle.kubernetes.operator.work.NextAction;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+
+public abstract class HttpResponseStep extends Step {
+  static final String RESPONSE = "httpResponse";
+
+  public HttpResponseStep(Step next) {
+    super(next);
+  }
+
+  @Override
+  public NextAction apply(Packet packet) {
+    return Optional.ofNullable(getResponse(packet)).map(r -> doApply(packet, r)).orElse(null);
```

Comment:
Looks like this code already handles no response.

Reply:
Agreed, although it returns null, which doesn't work. Fixing that as well.